view: fix api error handling

shwstppr opened the pull request:
Fixes #294 
![Screenshot from 2020-04-17 17-09-54](https://user-images.githubusercontent.com/153340/79565440-5bccfb00-80ce-11ea-8eb1-4e5dccceac89.png)


Fixes parsing invalid error object

shwstppr commented:
@rhtyd I've fixed that message comparison issue. I don't have a better solution right now for the whole notification thing.
Maybe we can add a parameter for `api` function call such as `notifyError` (with false as the default value) and then remove `this.$notification.error(...` code for API calls from the whole codebase where we are just simply showing error in notifications.
Although I feel better to not do it in TP milestone.

onitake commented:
I tested b3805f8 and it solved the issue reported in #294 . I haven't tested 944c5f6 yet.
Is there a specific reason why you changed `(x && y) || z` to `x ? y : z`?
A general dislike of the ternary operator notwithstanding, I think the ?: construct is easier to read - but YMMV.

shwstppr commented:
@onitake change was due to an internal review comment. I like `?:` myself, easier to understand.
Although, we might have to refactor the error handling more exhaustively as mentioned in previous comment.

rhtyd commented:
discuss - come up with a general purpose way of declaring errors?

shwstppr commented:
@rhtyd I've added global method and refactored some code (some more to be done).
Does this look okay or should we handle error notification in `src/api/index.js` itself with some flag in `api` method?

shwstppr commented on src/utils/plugins.js:
```diff
@@ -77,3 +78,35 @@ export const pollJobPlugin = {
+      }
+      notification.error({
+        message: msg,
+        description: error && error.response && 'x-description' in error.response.headers ? error.response.headers['x-description'] : error.message,
```
@rhtyd shoud this read, `description: desc`?
